conans/client/tools/system_pm.py

```diff
@@ -114,6 +116,17 @@ def install(self, packages, update=True, force=False):
             self.update()
         self._install_any(packages)
 
+    def _parse_packages_arch(self, packages):
+        if self._settings and cross_building(self._settings):
+            _, _, _, host_arch = get_cross_building_settings(self._settings)
```

Comment:
I do not fully understand that packages will be installed for the host_arch always, as by definition, the apt-get will be running in the build-machine, not the host. Will this break everything for users with ``build_requires`` packages that use ``SystemPackageTool``?

Reply:
Thinking again, you are right. I thought in the case where someone wants to cross-building some package and need a library from system. But sometimes users need a tool rather a library, which means the arch build.

We could detect if arch is not declared and use `build_arch` instead. WDYT?